### DISCUSSION TITLE
rtl (Right to Left) CSS support in the code

### DISCUSSION THREAD

**Discussion by dotanm** [ampache/ampache#2696] [Q&A] [answered]
Where can I find the references to the RTL languages support in the code? (Hebrew, Farsi, Arabic)

There are some modifications that need to be done in order to have more proper display of the interface when viewing it in RTL languages.

I tried editing several files, and the interface did not change, so I am asking for help here...

Thank you!

**Comment by lachlan-00** [comment]
@AshotN  and @kuzi-moto do you know anything? i do remember someone working on ampache in arabic before but i forget who that was.

**Reply by kuzi-moto** [reply]
I don't remember anyone working on this before. Also, I don't think there is anything in the CSS about this. I would probably look at editing `templates/base.css`. I'm not really sure what is needed to implement rtl though.

**Reply by dotanm** [reply]
Sorry for the late reply. I didn't get notified about your answers...

I found out that you need to create a file called rtl.css in the folder:
themes/ThemeName/templates/

At the bottom of this message are the lines I placed in the file, and it worked for me. It is not perfect (I cannot get the visualizer to show), but the main interface is fully functional.

Can it be placed in the main code source for future users? It is loaded only for languages that are defined as RTL, so it will not be relevant to users interface.

This is the rtl.css file I am working with:
```
body, select, input {
	direction: rtl;
}

#header .box-inside, .cel_rating, .star-rating.dynamic-star-rating {
	direction: ltr;
}

#header .box-content > input {
	direction: rtl;
}

.searchart {
	float: right;
}

div.jp-volume-bar-value, div.jp-volume-bar {
	direction: ltr;
}
```

**Reply by lachlan-00** [reply]
> Can it be placed in the main code source for future users?

Of course it can!

Can you make a pull request? I always prefer that people are given correct attribution for their efforts.

Just add the file in the pull and examples of the languages and I'll test it out. If you can give pictures or screens where you need help we can work it out. There might also be some people around who can help out as well on an open pull.

I'm glad this was easy enough to resolve for you so let's help others

**Comment by dotanm** [comment, marked as answer]
Create a file called rtl.css in the folder:
themes/ThemeName/templates/

This is the rtl.css file I am working with:
```
body, select, input {
	direction: rtl;
}

#header .box-inside, .cel_rating, .star-rating.dynamic-star-rating {
	direction: ltr;
}

#header .box-content > input {
	direction: rtl;
}

.searchart {
	float: right;
}

div.jp-volume-bar-value, div.jp-volume-bar {
	direction: ltr;
}
```
